docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+
+    Assert.True(messageReceived);
+}
+```
+
+This unit test checks that the `CatalogViewModel` publishes the `AddProduct` message in response to its `AddCatalogItemCommand` being executed. Because the `MessagingCenter` class supports multicast message subscriptions, the unit test can subscribe to the `AddProduct` message and execute a callback delegate in response to receiving it. This callback delegate, specified as a lambda expression, sets a boolean field that's used by the `Assert` statement to verify the behavior of the test.
+
+## Testing exception handling
+
+Unit tests can also be written that check that specific exceptions are thrown for invalid actions or inputs, as demonstrated in the following code example:
+
+```csharp
+[Fact]
+public void InvalidEventNameShouldThrowArgumentExceptionText()
+{
+    var behavior = new MockEventToCommandBehavior
+    {
+        EventName = "OnItemTapped"
+    };
+    var listView = new ListView();
+
+    Assert.Throws<ArgumentException>(() => listView.Behaviors.Add(behavior));
+}
+```
+
+This unit test will throw an exception because the `ListView` control does not have an event named `OnItemTapped`. The `Assert.Throws<T>` method is a generic method where `T` is the type of the expected exception. The argument passed to the `Assert.Throws<T>` method is a lambda expression that will throw the exception. Therefore, the unit test will pass provided that the lambda expression throws an `ArgumentException`.
+
+> [!TIP]
+> Avoid writing unit tests that examine exception message strings.
+
+Exception message strings might change over time, and so unit tests that rely on their presence are regarded as brittle.
+
+## Testing validation
+
+There are two aspects to testing the validation implementation: testing that any validation rules are correctly implemented and testing that the `ValidatableObject<T>` class performs as expected.
+
+Validation logic is usually simple to test, because it is typically a self-contained process where the output depends on the input. There should be tests on the results of invoking the `Validate` method on each property that has at least one associated validation rule, as demonstrated in the following code example:
+
+```csharp
+[Fact]
+public void CheckValidationPassesWhenBothPropertiesHaveDataTest()
+{
+    var mockViewModel = new MockViewModel();
+    mockViewModel.Forename.Value = "John";
+    mockViewModel.Surname.Value = "Smith";
+
+    bool isValid = mockViewModel.Validate();
+
+    Assert.True(isValid);
+}
+```
+
+This unit test checks that validation succeeds when the two `ValidatableObject<T>` properties in the `MockViewModel` instance both have data.
+
+As well as checking that validation succeeds, validation unit tests should also check the values of the `Value`, `IsValid`, and `Errors` property of each `ValidatableObject<T>` instance, to verify that the class performs as expected. The following code example demonstrates a unit test that does this:
```

Comment:
Why? This goes against what you stated earlier in this chapter. You said this earlier, which kind of reads as a contradiction:

> "Don't be tempted to make a unit test exercise more than one aspect of the unit's behavior."

Reply:
I think adding the additional explanation would be perfect here.